src/reporting/tezos/TezosConseilClient.ts

```diff
@@ -296,7 +296,43 @@ export namespace TezosConseilClient {
             maps.push({definition, content});
         }
 
-        return { contract: accountID, maps };
+        return { contract, maps };
+    }
+
+    /**
+     * Returns a value for a given key. The big map id is either provided as a paramter or the smallest of the possibly multiple big maps associated with the given contract address is used. Note that sometimes key values must be wrapped in quotes for key types that are non-numeric and not byte type.
+     * 
+     * @param serverInfo Conseil server connection definition.
+     * @param key Key to query for.
+     * @param contract Optional contract address to be used to identify an associated big map.
+     * @param mapIndex Optional big map index to query, but one of contract or mapIndex must be provided.
+     */
+    export async function getBigMapValueForKey(serverInfo: ConseilServerInfo, key: string, contract: string = '', mapIndex: number = -1): Promise<string> {
+        if (!contract.startsWith('KT1')) { throw new Error('Invalid address'); }
+        if (key.length < 1) { throw new Error('Invalid key'); }
+        if (mapIndex < 0 && contract.length === 0) { throw new Error('One of contract or mapIndex must be specified'); }
+
+        if (mapIndex < 0 && contract.length > 0) {
+            let ownerQuery = ConseilQueryBuilder.setLimit(ConseilQueryBuilder.blankQuery(), 1);
+            ownerQuery = ConseilQueryBuilder.addFields(ownerQuery, 'big_map_id');
+            ownerQuery = ConseilQueryBuilder.addPredicate(ownerQuery, 'account_id', ConseilOperator.EQ, [contract], false);
+            ownerQuery = ConseilQueryBuilder.addOrdering(ownerQuery, 'big_map_id', ConseilSortDirection.DESC);
+
+            const ownerResult = await getTezosEntityData(serverInfo, serverInfo.network, 'originated_account_maps', ownerQuery);
+
+            if (ownerResult.length < 1) { throw new Error(`Could not find any maps for ${contract}`); }
+            mapIndex = ownerResult[0];
+        }
+
+        let contentQuery = ConseilQueryBuilder.setLimit(ConseilQueryBuilder.blankQuery(), 1);
+        contentQuery = ConseilQueryBuilder.addFields(contentQuery, 'value');
+        contentQuery = ConseilQueryBuilder.addPredicate(contentQuery, 'key', ConseilOperator.EQ, [key], false);
```

Comment:
Is it not possible to automatically encode the key? I forget the details.

Reply:
not at this time, contract parser will need updates.